core/common/src/main/java/alluxio/ClientContext.java

```diff
@@ -36,6 +38,7 @@
  */
 public class ClientContext {
   private volatile AlluxioConfiguration mConf;
+  private volatile Map<String, AlluxioConfiguration> mPathConf = new HashMap<>();
```

Comment:
we will need to optimize the data structure for path conf lookup instead of a map

Reply:
also should we have this context (to resolve path defaults) as part of AlluxioConfiguration itself rather than a standalone type in clientcontext? that way we don't need to special case loading or updating path defaults from regular alluxio configuration.

Reply:
this depends on how we logically represent path level config:

option 1: map<path, map<key, value>>, which means for each path, we have a standalone AlluxioConfiguration instance holding the customized properties for that path, in this case, we cannot merge the path level configs into the cluster level configs because of property name conflicts.

option 2: path level configs have special property names, such as path.<path name>.<property key>=<property value>, in this case, we can merge them into cluster level configs.

I do not have a preference personally, what's your thought?